src/pal/src/numa/numa.cpp

```diff
@@ -140,71 +147,92 @@ BOOL
 NUMASupportInitialize()
 {
 #if HAVE_NUMA_H
-    if (numa_available() != -1)
+    numaHandle = dlopen("libnuma.so", RTLD_LAZY);
+    if (numaHandle == 0)
     {
-        struct bitmask *mask = numa_allocate_cpumask();
-        int numaNodesCount = numa_max_node() + 1;
-
-        g_possibleCpuCount = numa_num_possible_cpus();
-        g_cpuCount = 0;
-        g_groupCount = 0;
-
-        for (int i = 0; i < numaNodesCount; i++)
+        numaHandle = dlopen("libnuma.so.1", RTLD_LAZY);
```

Comment:
@janvorli, I was packaging a .NET application in a unikernel (bootable image) isolation, and found that if we want to include libnuma in the published disk image, we need to include one of these "symlinks" in addition to the actual binary: `libnuma.so.1.0.0`. In .NET 6's singlefilehost, libnuma (and liblttng-ust) are not statically compiled (yet, and maybe will not be in the future, since these two are soft dependencies?).

Since their ABI version hasn't changed in years (it is 1.0.0 everywhere I have looked), would adding a probing path with reordering: https://github.com/dotnet/runtime/commit/b60f1fabf1e07655350be4f1f36a4647edd0cbac be an acceptable change?